File: src/client/java/fr/hardel/asset_editor/client/javafx/lib/utils/BrowserUtils.java
package fr.hardel.asset_editor.client.javafx.lib.utils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class BrowserUtils {

    private static final Logger LOGGER = LoggerFactory.getLogger(BrowserUtils.class);

    public static boolean openBrowser(String url) {
        try {
            Runtime.getRuntime().exec(new String[]{"rundll32", "url.dll,FileProtocolHandler", url});
            return true;
        } catch (Exception e) {
            LOGGER.warn("Failed to open browser for {}: {}", url, e.getMessage());
            return false;
        }
    }

    private BrowserUtils() {
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/lib/utils/BrowserUtils.kt
package fr.hardel.asset_editor.client.compose.lib.utils

import org.slf4j.LoggerFactory

object BrowserUtils {

    private val logger = LoggerFactory.getLogger(BrowserUtils::class.java)

    fun openBrowser(url: String): Boolean = try {
        Runtime.getRuntime().exec(arrayOf("rundll32", "url.dll,FileProtocolHandler", url))
        true
    } catch (exception: Exception) {
        logger.warn("Failed to open browser for {}: {}", url, exception.message)
        false
    }
}
